src/jit/lclvars.cpp

```diff
@@ -235,6 +235,52 @@ void Compiler::lvaInitTypeRef()
 
     lvaInitArgs(&varDscInfo);
 
+#if FEATURE_FASTTAILCALL
+
+    //-------------------------------------------------------------------------
+    // Calculate the argument register usage.
+    //
+    // This will later be used for fastTailCall determination
+    //-------------------------------------------------------------------------
+
+    unsigned argRegCount      = 0;
+    unsigned floatingRegCount = 0;
+    unsigned stackArgCount    = 0;
+    size_t   stackSize        = 0;
+    unsigned compArgCount     = info.compArgsCount;
+
+    auto incrementRegCount = [&floatingRegCount, &argRegCount](LclVarDsc* varDsc) {
+        varDsc->IsFloatRegType() ? ++floatingRegCount : ++argRegCount;
+    };
```

Comment:
For amd64 Windows, a floating-point argument causes the corresponding integer argument to be skipped, and vice-versa. So for Windows amd64, shouldn't both be incremented no matter the type?

Also, what about HFA? In that case, IsFloatRegType() returns true. It seems like you might need to use lvHfaSlots() somehow, or get the actual size from lvExactSize. It's a little surprising that I couldn't find a member of LclVarDsc that return the number of registers used.